File: reducing-technical-debt-michael-feathers-course/techdebt/cs/scheduler/MailService.cs
using System;

namespace SchedulerCSharp
{
	public class MailService
	{		
		private static MailService instance;

		private MailService() 
		{
		}

		public static MailService Instance 
		{
			get 
			{
				if (instance == null)
					instance = new MailService();
				return instance;
			}
		}

		public virtual void SendMail(string address, string subject, string message) 
		{
			// this method really sends mail
		}

	}
}

File: reducing-technical-debt-michael-feathers-course/techdebt/cpp/scheduler/MailService.cpp


#include "MailService.h"


MailService MailService::instance;

MailService::MailService()
{
}

MailService *MailService::getInstance()
{
	return &instance;
}


void MailService::sendMail(const string& address, const string& subject, const string& message) 
{
}

